featuretools/entityset/entityset.py

```diff
@@ -1034,14 +1044,74 @@ def add_last_time_indexes(self, updated_dataframes=None):
+                    es_lti_dict[dataframe.ww.name] = lti_df
+                    es_lti_dict[dataframe.ww.name].name = 'last_time'
 
             explored.add(dataframe.ww.name)
-        self.reset_data_description()
 
+        # Store the last time index on the DataFrames
+        dfs_to_update = {}
+        for df in self.dataframes:
+            lti = es_lti_dict[df.ww.name]
+            if lti is not None:
+                if self.time_type == 'numeric':
+                    # --> don't totally understand why this is necessary when we can do the conversion normally?
+                    # --> need to make sure this works for dask and koalas
+                    if lti.dtype == 'datetime64[ns]':
+                        lti = lti.apply(lambda x: x.value)
+                    lti = ww.init_series(lti, logical_type='Double')
+                else:
+                    lti = ww.init_series(lti, logical_type='Datetime')
+
+                # Remove any existing last_time column to prevent collisions
+                lti.name = 'last_time'
+                if 'last_time' in df.columns:
+                    df.ww.pop('last_time')
+
+                # Add the new column to the DataFrame
+                if isinstance(df, dd.DataFrame):
+                    new_df = df.merge(lti.reset_index(), on=df.ww.index)
+                    new_df.ww.init(
+                        name=df.ww.name,
+                        index=df.ww.index,
+                        time_index=df.ww.time_index,
+                        logical_types=df.ww.logical_types,
+                        semantic_tags={col_name: tags - {'index', 'time_index'} for col_name, tags in df.ww.semantic_tags.items()},
+                        table_metadata=df.ww.metadata,
+                        column_metadata={col_name: col_schema.metadata for col_name, col_schema in df.ww.columns.items()},
+                        use_standard_tags=df.ww.use_standard_tags
+                    )
+
+                    new_idx = new_df[new_df.ww.index]
+                    new_idx.name = None
+                    new_df.index = new_idx
+                    dfs_to_update[df.ww.name] = new_df
+                elif is_instance(df, ks, 'DataFrame'):
+                    new_df = df.merge(lti, left_on=df.ww.index, right_index=True)
+
+                    new_df.ww.init(
+                        name=df.ww.name,
+                        index=df.ww.index,
+                        time_index=df.ww.time_index,
+                        logical_types=df.ww.logical_types,
+                        semantic_tags={col_name: tags - {'index', 'time_index'} for col_name, tags in df.ww.semantic_tags.items()},
+                        table_metadata=df.ww.metadata,
+                        column_metadata={col_name: col_schema.metadata for col_name, col_schema in df.ww.columns.items()},
+                        use_standard_tags=df.ww.use_standard_tags
```

Comment:
I'm not the biggest fan of this way of initializing Woodwork. It seems very likely to cause bugs/need updating every time Woodwork accessors get updated. 

But we can't initialize with a TableSchema here because the merge operations often change the dtypes, so the schema wouldn't be valid even after we went through the trouble of updating it to contain the last time index column. I played around with the idea of a function to allow initializing Woodwork with a partial schema (or maybe a partial columns dictionary?) while performing type inference on the columns without any schemas. I think this would also be useful for Woodwork's concat.

Reply:
> I played around with the idea of a function to allow initializing Woodwork with a partial schema

I like the idea of this type of behavior - use as much as we can and infer the rest. I think this might be useful in other situations beyond concat as well. Maybe we should add a couple issues - one to add this capability to Woodwork and a second to circle back and clean this up later?

Reply:
Sounds good, I'll create the issues!